File: test_backend_queries.js

const axios = require('axios');

async function test(q) {
    try {
        const res = await axios.get(`http://localhost:8000/api/parts?q=${encodeURIComponent(q)}`);
        console.log(`Query: "${q}" -> Results: ${res.data.length}`);
        if (res.data.length > 0) {
            console.log(`   Sample: ${res.data[0].part_code} - ${res.data[0].name}`);
        }
    } catch (e) {
        console.error(`Query: "${q}" -> Failed: ${e.message}`);
    }
}

async function run() {
    console.log('--- Testing Backend Search Patterns ---');
    await test('A0V');          // Code prefix
    await test('a0v');          // Case insensitive code
    await test('A0VDM50500');   // Full code
    await test('Thermistor');   // Description (exact case)
    await test('thermistor');   // Description (lowercase)
    await test('Termistore');   // Italian (should fail unless in DB)
    await test('ROLLER');       // Common part
    await test('roller');       // Case insensitive part
}

run();

File: test_backend_queries.py

import requests
import json

def test(q):
    try:
        url = f"http://localhost:8000/api/parts?q={q}"
        res = requests.get(url)
        data = res.json()
        print(f"Query: '{q}' -> Results: {len(data)}")
        if data:
            print(f"   Sample: {data[0]['part_code']} - {data[0]['name']}")
    except Exception as e:
        print(f"Query: '{q}' -> Failed: {e}")

print("--- Testing Backend Search Patterns ---")
test("A0V")
test("a0v")
test("A0VDM50500")
test("Thermistor")
test("thermistor")
test("ROLLER")
test("roller")
